src/main/java/com/blazemeter/jmeter/videostreaming/dash/Manifest.java

```diff
@@ -81,9 +86,13 @@ public Duration getMinimumUpdatePeriod() {
     return mpd.getMinimumUpdatePeriod();
   }
 
-  public long getReloadTimeMillis(Instant now) {
-    return Math
-        .max(mpd.getMinimumUpdatePeriod().minus(Duration.between(downloadTime, now)).toMillis(), 0);
+  public long getReloadTimeMillis(long segmentDurationMillis) {
+    Duration minUpdatePeriod = mpd.getMinimumUpdatePeriod();
+    // wait at least segment duration if minUpdatePeriod is 0 (or very small)
+    long maxIntervalTime = Math.max(minUpdatePeriod.toMillis(), segmentDurationMillis);
+    Instant now = Instant.now();
+
+    return Math.max(maxIntervalTime - Duration.between(lastDownLoadTime, now).toMillis(), 0);
```

Comment:
Both `minUpdatePeriod` & `now` could be inlined, due to them are only used once.

```suggestion
  public long getReloadTimeMillis(long segmentDurationMillis) {
    // wait at least segment duration if minUpdatePeriod is 0 (or very small)
    long maxIntervalTime = Math.max(mpd.getMinimumUpdatePeriod().toMillis(), segmentDurationMillis);
    return Math
        .max(maxIntervalTime - Duration.between(lastDownLoadTime, Instant.now()).toMillis(), 0);
  }
```